Automattic/sensei
# Add quiz migration

Resolves #5438 

## Proposed Changes

* Add a migration that migrates all the quiz data into the custom tables.
* Update the migration job scheduler to run multiple jobs, one after another.
* Refactor the migrations code a bit.

## Testing Instructions

1. Have some quiz entries.
1. Enable the feature: `add_filter( 'sensei_feature_flag_tables_based_progress', '__return_true' );`.
1. Go to Sensei LMS -> Tools.
1. Run `Migrate comment-based progress`.
1. It should first migrate the student progress progress and then run the quiz migration.
1. Check the quiz tables and verify that the data is correct.
1. To rerun the migration, truncate the tables and delete the `sensei_migrated_quiz_last_comment_id` option.

## Pre-Merge Checklist
